lighthouse/environment/src/lib.rs

```diff
@@ -38,6 +35,21 @@ const LOG_CHANNEL_SIZE: usize = 2048;
 /// The maximum time in seconds the client will wait for all internal tasks to shutdown.
 const MAXIMUM_SHUTDOWN_TIME: u64 = 15;
 
+/// Configuration for logging.
+/// Background file logging is disabled if one of:
```

Comment:
Perhaps this `enum` expresses this relationship and would be a little easier to read:

```rust
enum FileLogging {
  Disabled,
  Enabled {
    path: PathBuf,
    max_log_size: u64,
    max_log_number: usize,
  }
}
```